backend/api/serializers/computetask.py

```diff
@@ -258,17 +282,27 @@ def create(self, validated_data):
         outputs = validated_data.pop("outputs")
 
         compute_task = super().create(validated_data)
+        input_kinds = {algo_input.identifier: algo_input.kind for algo_input in compute_task.algo.inputs.all()}
 
         for order, data_sample in enumerate(data_samples):
             TaskDataSamples.objects.create(compute_task=compute_task, data_sample=data_sample, order=order)
 
         for position, input in enumerate(inputs):
-            ComputeTaskInput.objects.create(
+            task_input = ComputeTaskInput.objects.create(
                 channel=compute_task.channel,
                 task=compute_task,
                 position=position,
                 **input,
             )
+            # task input asset could be known during task registration
+            # or could be resolved later if it does not exist yet
```

Comment:
There is a catch here: if you add a task with already DONE parents, we already have all the inputs.
It's probably better to only resolve links at query time in this context (we can afford the additional cost since inputs should eventually be a task subresource).

Reply:
Indeed this issue is catched by the unit tests.

Reply:
I am not sure this invalidate having `ComputeTaskInputAsset` though.

Reply:
I think I misunderstood the flow: this is in the `create` method, so during processing of the task event right? What about resources registered afterward, how are `ComputeTaskInput` objects created for them? Or is this only happening when building the API response?

Reply:
This should be covered by the compute task output asset [sync](https://github.com/Substra/substra-backend/pull/458/files#diff-50b50fd37d33a5998bcd7a185d86f4a84641f351a3f9c3198adbdd5035e4b141R234).

Reply:
Thanks for the clarification, that's the part I missed 👍